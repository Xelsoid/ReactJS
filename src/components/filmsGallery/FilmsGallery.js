import PropTypes from "prop-types";
import React from 'react';

import './filmsGallery.scss';
import FilmsGalleryCard from './FilmsGalleryCard';

const FilmsGallery = ({ requestedFilms }) => {
    const {data} = requestedFilms;
    return (
      <div className="film-gallery">
        {data.map(film => (
          <div key={film.id} className="film-gallery--column">
            <FilmsGalleryCard film={film} />
          </div>
        ))}
      </div>
    )
};

FilmsGallery.defaultProps = {
    requestedFilms: null
};

FilmsGallery.propTypes = {
    requestedFilms: PropTypes.instanceOf(Object),
};

export default FilmsGallery;
