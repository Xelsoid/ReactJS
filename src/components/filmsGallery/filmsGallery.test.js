import React from 'react';
import { FilmsGallery } from './FilmsGallery';
import requestedFilms from '../../mockedData/MOCKED_DATA';


describe('FilmsGallery', () => {
  it('renders correctly', () => {
    const component = shallow(
      <FilmsGallery movies={requestedFilms.data} chooseFilm={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
