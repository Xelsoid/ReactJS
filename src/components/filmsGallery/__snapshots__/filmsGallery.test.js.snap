// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsGallery renders correctly 1`] = `
<div
  className="film-gallery"
>
  <div
    className="film-gallery--column"
    key="337167"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 55000000,
          "genres": Array [
            "Drama",
            "Romance",
          ],
          "id": 337167,
          "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
          "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
          "release_date": "2018-02-07",
          "revenue": 136906000,
          "runtime": 106,
          "tagline": "Don't miss the climax",
          "title": "Fifty Shades Freed",
          "vote_average": 6.1,
          "vote_count": 1195,
        }
      }
    />
    <NavLink
      to="/description/Fifty Shades Freed"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={337167}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="269149"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 150000000,
          "genres": Array [
            "Animation",
            "Adventure",
            "Family",
            "Comedy",
          ],
          "id": 269149,
          "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
          "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
          "release_date": "2016-02-11",
          "revenue": 1023784195,
          "runtime": 108,
          "tagline": "Welcome to the urban jungle.",
          "title": "Zootopia",
          "vote_average": 7.7,
          "vote_count": 6795,
        }
      }
    />
    <NavLink
      to="/description/Zootopia"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={269149}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="181808"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 200000000,
          "genres": Array [
            "Fantasy",
            "Adventure",
            "Science Fiction",
          ],
          "id": 181808,
          "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
          "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
          "release_date": "2017-12-13",
          "revenue": 1325937250,
          "runtime": 152,
          "tagline": "The Saga Continues",
          "title": "Star Wars: The Last Jedi",
          "vote_average": 7.1,
          "vote_count": 4732,
        }
      }
    />
    <NavLink
      to="/description/Star Wars: The Last Jedi"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={181808}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="284054"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 200000000,
          "genres": Array [
            "Action",
            "Adventure",
            "Fantasy",
            "Science Fiction",
          ],
          "id": 284054,
          "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
          "poster_path": "https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
          "release_date": "2018-02-13",
          "revenue": 1245257672,
          "runtime": 134,
          "tagline": "Long live the king",
          "title": "Black Panther",
          "vote_average": 7.3,
          "vote_count": 3788,
        }
      }
    />
    <NavLink
      to="/description/Black Panther"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={284054}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="354912"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Comedy",
            "Family",
            "Animation",
          ],
          "id": 354912,
          "overview": "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
          "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          "release_date": "2017-10-27",
          "revenue": 700920729,
          "runtime": 105,
          "tagline": "The celebration of a lifetime",
          "title": "Coco",
          "vote_average": 7.8,
          "vote_count": 3619,
        }
      }
    />
    <NavLink
      to="/description/Coco"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={354912}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="333339"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "revenue": 0,
          "runtime": 140,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
          "vote_average": 8.1,
          "vote_count": 617,
        }
      }
    />
    <NavLink
      to="/description/Ready Player One"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={333339}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="338970"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 94000000,
          "genres": Array [
            "Action",
            "Adventure",
          ],
          "id": 338970,
          "overview": "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
          "poster_path": "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
          "release_date": "2018-03-08",
          "revenue": 126025000,
          "runtime": 118,
          "tagline": "Her legend begins",
          "title": "Tomb Raider",
          "vote_average": 6.2,
          "vote_count": 817,
        }
      }
    />
    <NavLink
      to="/description/Tomb Raider"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={338970}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="284053"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 180000000,
          "genres": Array [
            "Action",
            "Adventure",
            "Fantasy",
          ],
          "id": 284053,
          "overview": "Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the prophecy of destruction to his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
          "poster_path": "https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg",
          "release_date": "2017-10-25",
          "revenue": 854229371,
          "runtime": 130,
          "tagline": "No Hammer. No Problem.",
          "title": "Thor: Ragnarok",
          "vote_average": 7.4,
          "vote_count": 5349,
        }
      }
    />
    <NavLink
      to="/description/Thor: Ragnarok"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={284053}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="321612"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 160000000,
          "genres": Array [
            "Family",
            "Fantasy",
            "Romance",
          ],
          "id": 321612,
          "overview": "A live-action adaptation of Disney's version of the classic tale of a cursed prince and a beautiful young woman who helps him break the spell.",
          "poster_path": "https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg",
          "release_date": "2017-03-16",
          "revenue": 1263521126,
          "runtime": 129,
          "tagline": "Be our guest.",
          "title": "Beauty and the Beast",
          "vote_average": 6.8,
          "vote_count": 7861,
        }
      }
    />
    <NavLink
      to="/description/Beauty and the Beast"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={321612}
        title="Show Description"
      />
    </NavLink>
  </div>
  <div
    className="film-gallery--column"
    key="399055"
  >
    <FilmsGalleryCard
      film={
        Object {
          "budget": 19500000,
          "genres": Array [
            "Drama",
            "Fantasy",
            "Romance",
          ],
          "id": 399055,
          "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
          "poster_path": "https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg",
          "release_date": "2017-12-01",
          "revenue": 185545281,
          "runtime": 123,
          "tagline": "A Fairy Tale for Troubled Times",
          "title": "The Shape of Water",
          "vote_average": 7.3,
          "vote_count": 3200,
        }
      }
    />
    <NavLink
      to="/description/The Shape of Water"
    >
      <Button
        btnClass="btn--primary"
        callback={[Function]}
        disabled={false}
        id={399055}
        title="Show Description"
      />
    </NavLink>
  </div>
</div>
`;
